src/app/shared/services/translation.service.ts

```diff
@@ -67,57 +67,60 @@ setIndex() {
 }
 
 translate(text: string) {
-    if (typeof text !== 'string') {
-        return text;
-    }
-
-    const length = this.dictionary.value.length;
-
-    return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length) || text;
-
+  if (typeof text !== 'string') {
+      return text;
+  }
+
+  const length = this.dictionary.value.length;
+
+  if (this.currentLanguage === 'hi') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation') || text;
+  } else if (this.currentLanguage === 'or') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation2') || text;
+  } else {
+      return text;
+  }
```

Comment:
```suggestion

let tranlsationKey;

  if (this.currentLanguage === 'hi') {
      translationKey = 'translation';
  } else if (this.currentLanguage === 'or') {
      translationKey = 'translation2';
  } else {
      return text;
  }
  
  return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, translationKey) || text;
```

Avoid repeating code where possible